sdk/nodejs/cosmosdb/zMixins.ts

```diff
@@ -0,0 +1,203 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as pulumi from "@pulumi/pulumi";
+import { Account } from "./account";
+
+import * as appservice from "../appservice";
+import * as core from "../core";
+import * as util from "../util";
+
+interface CosmosBindingDefinition extends appservice.BindingDefinition {
+    /**
+     * The name of the property in the context object to bind the actual items to.
+     */
+    name: string;
+
+    /**
+     * The type of a Cosmos DB binding. Must be 'cosmosDBTrigger'.
+     */
+    type: "cosmosDBTrigger";
+
+    /**
+     * The direction of the binding. We only support collection items being inputs to functions.
+     */
+    direction: "in";
+
+    /**
+     * The name of the database we are subscribing to.
+     */
+    databaseName: pulumi.Input<string>;
+
+    /**
+     * The name of the collection inside the database we are subscribing to.
+     */
+    collectionName: pulumi.Input<string>;
+
+    /**
+     * When set, it adds a prefix to the leases created in the Lease collection for this Function, effectively allowing 
+     * two separate Azure Functions to share the same Lease collection by using different prefixes.
+     */
+    leaseCollectionPrefix: pulumi.Input<string>;
+
+    /**
+     * When set to true, the leases collection is automatically created when it doesn't already exist. 
+     * The default value is false.
```

Comment:
it says default is false, but passes 'true' below.

Reply:
If we don't specify it in binding properties, Azure will assume `false`. That's why we pass `true` below. Too confusing?

Reply:
gotcha.  it is a bit confusing.  so we should just explain what's going on in the code (either here or at the poitn you pass in 'true').  we should also state why it's appropriate to pass in 'true' .